_posts/2019-02-27-stripe-connect-part-1.md

```diff
@@ -0,0 +1,187 @@
+- Once in that endpoint, we make a POST request to Stripe to finish the account creation
+- If the response was successful, we save in our database the `stripe_user_id`  that we received and then we show a success message to the user.
+
+### Create your own Stripe account
+If you don't have a Stripe account yet you can create one [here](https://dashboard.stripe.com/register?redirect=%2Fconnect%2Foverview).
+For now you'll only need it to access to your [settings](https://dashboard.stripe.com/account/applications/settings) and [credentials](https://dashboard.stripe.com/account/apikeys).
+
+### Add a "Connect with Stripe" button
+Users should have a [link](https://stripe.com/docs/connect/express-accounts#integrating-oauth) to create their Stripe account.
+For the UI we can use one of the [designed buttons](https://stripe.com/newsroom/brand-assets) that Stripe offers.
+
+```ruby
+# app/views/users/settings.html.erb
+
+<%= link_to image_tag("stripe_connect_button.png"), stripe_connect_link %>
+```
+
+```ruby
+# app/helpers/users_helper.rb
+
+def stripe_connect_link
+  stripe_url = "https://connect.stripe.com/express/oauth/authorize"
+  redirect_uri = stripe_connect_url
+  client_id = ENV["STRIPE_CLIENT_ID"]
+
+  "#{stripe_url}?redirect_uri=#{redirect_uri}&client_id=#{client_id}"
+end
+```
+
+### Add an endpoint to be redirected after the account setup
+
+In the URL for the "Connect with Stripe" button we specified that after the user completes the Stripe account setup, it will [redirect](https://stripe.com/docs/connect/express-accounts#redirected) to `stripe_connect_url`, so let's create that endpoint.
+
+You should also include the full link to this endpoint in your [settings](https://dashboard.stripe.com/account/applications/settings).
+
+```ruby
+# routes.rb
+
+get "stripe/connect", to: "stripe#connect", as: :stripe_connect
+```
+
+```ruby
+# app/controllers/stripe_controller.rb
+
+class StripeController < ApplicationController
+  def connect
+  end
+end
+```
+
+### Make a POST request to Stripe to finish the account setup
+At this point we have the necessary params to make the final [request](https://stripe.com/docs/connect/express-accounts#complete-express-connection) to complete the account connection. We're using [httparty](https://github.com/jnunemaker/httparty) for that, but you can do it with [Net::HTTP
+](https://ruby-doc.org/stdlib-2.6.1/libdoc/net/http/rdoc/Net/HTTP.html) too.
+
+```ruby
+# app/controllers/stripe_controller.rb
+
+class StripeController < ApplicationController
+  def connect
+    response = HTTParty.post("https://connect.stripe.com/oauth/token",
+      query: {
+        client_secret: ENV["STRIPE_SECRET_KEY"],
+        code: params[:code],
+        grant_type: "authorization_code"
+      }
+    )
+  end
+end
+```
+### Add `stripe_user_id` to our database
+
+We should have the `stripe_user_id` column in our users table so we can connect the Stripe accounts to our User objects.
```

Comment:
Can you elaborate this a little bit? How/why would we need to connect both?

Reply:
@cleicar Sure, just added some extra details